File: pmul/android/aplicacion-elecciones/app/src/main/java/com/example/aplicacion_elecciones/AccountManager.java
package com.example.aplicacion_elecciones;

import com.example.aplicacion_elecciones.database.entities.Usuario;
import com.example.aplicacion_elecciones.database.entities.UsuarioDAO;

public class AccountManager
{
    private static AccountManager instance;

    private Usuario currentUser;

    private AccountManager()
    {
    }

    public static AccountManager getInstance()
    {
        if (instance == null)
        {
            instance = new AccountManager();
        }

        return instance;
    }

    public boolean login(String nif, String password)
    {
        currentUser = UsuarioDAO.validateUser(nif, password);

        return currentUser != null;
    }

    public boolean logout()
    {
        currentUser = null;
        return true;
    }

    public boolean isLogged()
    {
        return currentUser != null;
    }

    public Usuario getCurrentUser()
    {
        return currentUser;
    }
}

File: pmul/aplicacion-elecciones-kotlin/app/src/main/java/com/example/aplicacion_elecciones_kotlin/AccountManager.kt
package com.example.aplicacion_elecciones_kotlin

import com.example.aplicacion_elecciones_kotlin.database.entities.Usuario
import com.example.aplicacion_elecciones_kotlin.database.entities.UsuarioDAO

object AccountManager
{
    private var currentUser: Usuario? = null

    fun login(nif: String, password: String): Boolean
    {
        currentUser = UsuarioDAO.validateUser(nif, password)

        return currentUser != null
    }

    fun logout(): Boolean
    {
        currentUser = null
        return true
    }

    val isLogged: Boolean
        get() = currentUser != null
}
